sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -749,6 +790,48 @@ public void testSinkWithSendErrors() throws Throwable {
   }
 
   @Test
+  public void testUnboundedSourceStartReadTime() {
+
+    if (new ConsumerSpEL().hasOffsetsForTimes()) {
+      int numElements = 1000;
+      int startTime = numElements / 20 / 2; // half elements
```

Comment:
Looks confusing: why is start time derived from elements? What units is time measured in?